From d37315a77cdac769888d7cb3dcc5ceabe2719a20 Mon Sep 17 00:00:00 2001
From: nxa15299 <Heinz.Wrobel@nxp.com>
Date: Mon, 12 Feb 2018 07:52:20 +0100
Subject: [PATCH 1/2] u-boot-qoriq: VID code blocked interrupts

This is not a good idea as other necessary functions may depend on them.
The problem solved here is that U-Boot locked up during the VID code.

Signed-off-by: nxa15299 <Heinz.Wrobel@nxp.com>
---
 board/freescale/common/vid.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/board/freescale/common/vid.c b/board/freescale/common/vid.c
index a7835e45de..d66957ad6f 100644
--- a/board/freescale/common/vid.c
+++ b/board/freescale/common/vid.c
@@ -518,7 +518,6 @@ int vid_set_mv_limits(int absmax,
 
 int adjust_vdd(ulong vdd_override)
 {
-	int re_enable = disable_interrupts();
 #if defined(CONFIG_FSL_LSCH2) || defined(CONFIG_FSL_LSCH3)
 	struct ccsr_gur *gur = (void *)(CONFIG_SYS_FSL_GUTS_ADDR);
 #else
@@ -700,9 +699,6 @@ int adjust_vdd(ulong vdd_override)
 	else
 		ret = -1;
 exit:
-	if (re_enable)
-		enable_interrupts();
-
 	i2c_multiplexer_select_vid_channel(I2C_MUX_CH_DEFAULT);
 
 	return ret;
-- 
2.15.1

